uint32 width
uint32 height
int8[] data
---
int8[] cleanedData